src/main/java/training/chessington/model/pieces/Pawn.java

```diff
@@ -80,4 +80,9 @@ private Move getDoubleMove(Coordinates currentSquare, Board board) {
         return allowedMoves;
     }
 
+    protected boolean moveIsValid(Move move) {

```

Comment:
NB this method doesn't work correctly for capturing moves (it would disallow them).